src/python/pants/backend/awslambda/python/rules.py

```diff
@@ -33,27 +39,48 @@ class PythonAwsLambdaFieldSet(PackageFieldSet):
     complete_platforms: PythonFaaSCompletePlatforms
     output_path: OutputPathField
     environment: EnvironmentField
+    layout: PythonFaaSLayoutField
 
 
 @rule(desc="Create Python AWS Lambda", level=LogLevel.DEBUG)
 async def package_python_awslambda(
     field_set: PythonAwsLambdaFieldSet,
 ) -> BuiltPackage:
+    layout = PythonFaaSLayout(field_set.layout.value)
+
+    if layout is PythonFaaSLayout.LAMBDEX:
+        return await Get(
+            BuiltPackage,
+            BuildLambdexRequest(
+                address=field_set.address,
+                target_name=PythonAWSLambda.alias,
+                complete_platforms=field_set.complete_platforms,
+                runtime=field_set.runtime,
+                handler=field_set.handler,
+                output_path=field_set.output_path,
+                include_requirements=field_set.include_requirements.value,
+                script_handler=None,
+                script_module=None,
+                # The AWS-facing handler function is always lambdex_handler.handler, which is the
+                # wrapper injected by lambdex that manages invocation of the actual handler.
+                handler_log_message="lambdex_handler.handler",
+            ),
+        )
+
     return await Get(
         BuiltPackage,
-        BuildLambdexRequest(
+        BuildPythonFaaSRequest(
             address=field_set.address,
             target_name=PythonAWSLambda.alias,
             complete_platforms=field_set.complete_platforms,
             runtime=field_set.runtime,
             handler=field_set.handler,
+            layout=layout,
             output_path=field_set.output_path,
             include_requirements=field_set.include_requirements.value,
-            script_handler=None,
-            script_module=None,
-            # The AWS-facing handler function is always lambdex_handler.handler, which is the
-            # wrapper injected by lambdex that manages invocation of the actual handler.
-            handler_log_message="lambdex_handler.handler",
+            # this isn't built or run via lambdex, but the name is arbitrary, so we stick with the
```

Comment:
I think this is a good opportunity to get rid of this name, since we're churning anyway. It'll be harder to do later.

Reply:
But also, are we sure we want this at all? Why not use the handler name the user configured directly?

Reply:
> But also, are we sure we want this at all? Why not use the handler name the user configured directly?

Yes, I think so. The `BUILD` file target _must_ contain (most of) the entry point (at least the file/module part of `entry_point="path/to.py:handler"`), in order to compute dependencies/build the package, and it's annoying to have to carefully reproduce that elsewhere (`handler: 'parent.modules.path.to.handler'`), and synthesising a fixed module to incorporate the info pants already knows avoids that.

For instance, in our repo, we use AWS CDK, and have a generic `PythonLambda` construct where the basic "get the code running" config is just pointing it to the package (e.g. `foo.bar/target.zip`), because it can just set `handler: 'lambdex_handler.handler'` to get things running. I think this is really nice!

(There's obviously often per-lambda config required like any extra env vars or permissions, but CDK/the infra configuration is the source of truth there, it's not just directly duplicated from the `BUILD` files.)

> I think this is a good opportunity to get rid of this name, since we're churning anyway. It'll be harder to do later.

Sure, I'm happy to choose another name, e.g. some options:

- `lambda_function`: https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html suggests AWS configures `lambda_function.lambda_handler` by default, when creating a Python function using the console
- `index`: using "inline code" in AWS's CDK uses `index` as the file name (https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html#handler), but... the `handler` property always needs to be set
- `pants_lambda` or `__pants_lambda` or some other pants-y name

I think we can also make this customisable in future.

Reply:
I've gone with `lambda_function`, but more than happy to change. (Thought of another option too: `main` to be consistent with GCF.)